x-pack/dockerlogbeat/pipelinemanager/clientLogReader_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemanager
+
+import (
+	"testing"
+
+	"github.com/docker/docker/daemon/logger"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/x-pack/dockerlogbeat/pipelinemock"
+	"github.com/elastic/beats/x-pack/dockerlogbeat/pipereader"
+)
+
+func TestNewClient(t *testing.T) {
+	mockConnector := &pipelinemock.MockPipelineConnector{}
+	client := createNewClient(t, mockConnector)
+	// ConsumePipelineAndSent is what does the actual reading and sending.
+	// After we spawn this goroutine, we wait until we get something back
+	go client.ConsumePipelineAndSend()
+	event := testReturnAndClose(t, mockConnector, client)
+	assert.Equal(t, event.Fields["message"], "This is a log line")
+
+}
```

Comment:
We should check that the go-routine correctly finishes or block indefinitely (go kills tests by default after 10min).

Do we need `testReturnAndClose` to close the client? It breaks 'symmetry' (create and close split between functions).

How about
```
var wg sync.WaitGroup
defer wg.Wait()  // block test shutdown until created go-routines have finished

client := createNewTestClient(t, mockConnector)
defer client.Close()

wg.Add(1)
go func(){
  defer wg.Done()
  client.ConsumePipelineAndSend()
}()

event := testWaitEvent(t, mockConnector, client)
...
```

Assuming we always want to run the go-routine we can also write setup+teardown support like this:

```
func setupTestClient(t *testing.T, ...) (*ClientLogger, func()) {
  ...
  client, err := newClientFromPipeline(...)
  assert.NoError(t, err)

  var wg sync.WaitGroup
  wg.Add(1)
  go func() {
    defer wg.Done()
    client.ConsumePipelineAndSend()
  }()

  return client, func() {
    client.Close()
    wg.Wait()
  }
}

func TestX(t * testing.T) {
  client, teardown := setupTestClient(...)
  defer teardown()

  // run test
}
```